lib/core/options/builders.js

```diff
@@ -0,0 +1,483 @@
+					// The other cases have hardcoded class names
+					const element = thing.getAuthorElement();
+					return element.classList.contains(data.attr);
+				}
+			},
+		},
+		domain: {
+			name: 'Link domain name',
+			defaultTemplate(dom) {
+				return { type: 'domain', patt: dom || '' };
+			},
+			fields: [
+				'post links to the domain ',
+				{ type: 'text', id: 'patt', validator: RegExp },
+			],
+			evaluate(thing, data) {
+				const domain = thing.getPostDomain();
+				if (!domain) {
+					return false;
+				}
+				const pattern = new RegExp(`^(${data.patt})$`, 'i');
+				return pattern.test(domain);
+			},
+		},
+		dow: {
+			name: 'Day of week',
+			defaultTemplate() {
+				return { type: 'dow', days: [] };
+			},
+			fields: [
+				'current day of the week is ',
+				{
+					type: 'checkset', id: 'days',
+					// Uses same 3 letter names as
+					// .toLocaleDateString('en-US', {weekday: 'short'}))
+					items: 'Sun,Mon,Tue,Wed,Thu,Fri,Sat'.split(','),
+				},
+			],
+			evaluate(thing, data) {
+				// duplicating because I was having issues with accessing a variable before it was assigned
+				const dayList = 'Sun,Mon,Tue,Wed,Thu,Fri,Sat'.split(',');
+				// Get 3 letter name
+				const currentDOW = dayList[new Date().getDay()];
+
+				// At the time of writing Safari doesn't support the toLocaleDateString
+				// const currentDOW = new Date().toLocaleDateString('en-US', {weekday: 'short'});
+				return data.days.includes(currentDOW);
+			},
+		},
+		linkFlair: {
+			name: 'Link flair',
+			defaultTemplate(patt) {
+				return { type: 'linkFlair', patt: patt || '' };
+			},
+			fields: [
+				'post has link flair matching ',
+				{ type: 'text', id: 'patt', validator: RegExp },
+			],
+			evaluate(thing, data) {
+				const text = thing.getPostFlairText();
+				if (!text) {
+					return false;
+				}
+				const pattern = new RegExp(`^(${data.patt})$`, 'i');
+				return pattern.test(text.trim());
+			},
+		},
+		userFlair: {
+			name: 'User flair',
+			defaultTemplate(patt) {
+				return { type: 'userFlair', patt: patt || '' };
+			},
+			fields: [
+				'author of this post has flair matching ',
+				{ type: 'text', id: 'patt', validator: RegExp },
+			],
+			evaluate(thing, data) {
+				const text = thing.getUserFlairText();
+				if (!text) {
+					return false;
+				}
+				const pattern = new RegExp(`^(${data.patt})$`, 'i');
+				return pattern.test(text);
+			},
+		},
+		postType: {
+			name: 'Post type',
+			defaultTemplate(kind) {
+				return { type: 'postType', kind: kind || 'link' };
+			},
+			fields: [
+				'post is a ', {
+					// id: 'type' results in a collsion
+					type: 'select', id: 'kind',
+					options: [
+						['link post', 'link'],
+						['self post', 'self'],
+						// 'comment',
+					],
+				}, '.',
+			],
+			evaluate(thing, data) {
+				switch (data.kind) {
+					case 'comment': return thing.isComment();
+					case 'link':    return thing.isSelfPost();
+					case 'self':    return this.isLinkPost();
```

Comment:
`this` -> `thing`

Also these have always been backwards, apparently: `case 'link':    return thing.isSelfPost();`